broker/lib/controllers/FabrikBaseController.js

```diff
@@ -30,6 +33,67 @@ class FabrikBaseController extends BaseController {
     return _.get(config, 'broker_name', 'service-fabrik-broker');
   }
 
+  handleWithResourceLocking(func, operationType, lastOperationCall) {
+    return (req, res, next) => {
+      return Promise.try(() => {
+          if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR && config.enable_service_fabrik_v2) {
```

Comment:
code can be further made more readable if you modularize the method `handleWithResourceLocking` this way:
```
handleWithResourceLocking(func, operationType){
     return (req, res, next) => {
          let resourceLocked = false;
          return this._lockResource(req)
                     .tap(() => resourceLocked = true)
                     .then(()=> this._handle(req,res))
                     .then(()=>this._unlockIfReqfailed(req, res)) //This can have generic logic to check response code !=200
                     .catch(ContinueWithNext, () => {
                                  _.set(req, 'params_copy', req.params);
                                  return process.nextTick(next);
                      })
                     .catch((err)=> resourceLocked ? this._unlockIfReqfailed(err, req, res): next(err))
                       //_unlockIfReqfailed - common method which handles both error and res.statuscode != 202 
     }
}
```
The above code summarizes all the review comments provided for this method & its existing inner method(read below). Havent provided impl details of the inner methods, but their names should make it intuitive to see what should go in them.

Reply:
Done!